File: bookstore_backend/src/main/java/com/bookstore/services/VoucherService.java
package com.bookstore.services;

import org.springframework.data.domain.Pageable;

import com.bookstore.common.response.PageResponse;
import com.bookstore.dto.voucher.CreateVoucherRequest;
import com.bookstore.dto.voucher.VoucherResponse;

public interface VoucherService {
    PageResponse<VoucherResponse> getAllVouchers(Pageable pageable);

    PageResponse<VoucherResponse> getMyVouchers(int userId, Pageable pageable);

    VoucherResponse createVoucher(CreateVoucherRequest request);
}

File: bookstore_frontend/src/features/vouchers/services/voucherService.ts
import { apiClient } from '../../../lib/apiClient';
import type { ApiResponse, PageResponse } from '../../../types/api/common';
import type { CreateVoucherRequest, VoucherResponse } from '../../../types/api/voucher';

export const voucherService = {
  getMyVouchers: (page = 0, size = 20) =>
    apiClient.get<ApiResponse<PageResponse<VoucherResponse>>>('/api/vouchers/me', {
      params: { page, size, sort: 'voucherId,desc' },
    }),

  getAdminVouchers: (page = 0, size = 20) =>
    apiClient.get<ApiResponse<PageResponse<VoucherResponse>>>('/api/admin/vouchers', {
      params: { page, size, sort: 'voucherId,desc' },
    }),

  createVoucher: (request: CreateVoucherRequest) =>
    apiClient.post<ApiResponse<VoucherResponse>>('/api/admin/vouchers', request),
};
